src/MinhaCompra/src/views/Home.js

```diff
@@ -1,54 +1,90 @@
-import React, { useState, useEffect, useReducer } from 'react';
-import { StyleSheet, Text, View, Entypo } from 'react-native';
+import React, { useState, useEffect } from 'react';
+import { ScrollView, StyleSheet, Text, View } from 'react-native';
 import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
 import SearchBar from '../components/SearchBar';
 import Lists from '../components/Lists';
-import { consultaListaDeCompra, criaListaDeCompra } from "../services/DataService";
-import ButtonFab from '../components/ButtonFab';
+import { consultaListaDeCompra, criaListaDeCompra, limparListaDeCompra } from "../services/DataService";
 
 const Tap = createBottomTabNavigator()
 
 export default function Home() {
-  const [lists, setLists] = useState()
+  const [defaultLists, setDefaultLists] = useState()
+  const [filteredList, setFilteredList] = useState()
 
   useEffect(() => {
 
-    const mockedLists = [
-    { id: 0, id_produto: 234, valor_total: 50.00, cpf: 43205987654, nome_lista: 'Março' },
-    { id: 1, id_produto: 2343, valor_total: 55.00, cpf: 12305987654, nome_lista: 'Abril' },
-    { id: 2, id_produto: 23232, valor_total: 60.00, cpf: 22305987654, nome_lista: 'Maio' },
-    ];
+    // const mockedLists = [
+    //   { id: 0, id_produto: 234, valor_total: 50.00, cpf: 43205987654, nome_lista: 'Março' },
+    //   { id: 1, id_produto: 2343, valor_total: 55.00, cpf: 12305987654, nome_lista: 'Abril' },
+    //   { id: 2, id_produto: 23232, valor_total: 60.00, cpf: 22305987654, nome_lista: 'Maio' },
+    // ];
+
+    // criaListaDeCompra({
+    //   id_produto: 2,
+    //   valor_total: 10.0,
+    //   cpf: 40028922,
+    //   nome_lista: "Junho"
+    // })
+    //   .then((dados) => {
+    //     console.log(dados);
+    //   })
+    //   .catch((e) => console.log("error:", e));
 
     consultaListaDeCompra()
       .then(dados => {
-        setLists(dados)
+        console.log(dados)
+        setDefaultLists(dados)
       })
-      .catch((e) => e);
+      .catch((e) => console.log(`error: ${e}`));
   }, []);
 
+  const searchFilterFunction = async(text) => {
+    if(text) {
+      const newList = defaultLists.filter(item => {
+        const listItem = item.nome_lista.toLowerCase()
+        const inputText = text.toLowerCase()
+
+        return listItem.indexOf(inputText) > -1;
+      })
+      setFilteredList(newList)   
+
+    }
+  }
+
+  const handleOnChangeText = (event) => {
+    if(event) {
+      searchFilterFunction(event)
+    } else {
+      setFilteredList(defaultLists)
```

Comment:
Verificação necessária pra se o valor for false, setar o valor de filteredLists como defaultLists. Por mais que tenha a verificação no map das listas, essa é para o state e não o map.